src/plone/restapi/services/querystringsearch/get.py

```diff
@@ -73,14 +72,16 @@ class QuerystringSearchPost(Service):
 
     def reply(self):
         querystring_search = QuerystringSearch(self.context, self.request)
-        return querystring_search(data=json_body(self.request))
+        return querystring_search()
 
 
 class QuerystringSearchGet(Service):
     """Returns the querystring search results given a p.a.querystring data."""
 
     def reply(self):
+        # We need to copy the JSON query parameters from the querystring
+        # into the request body, because that's where other code expects to find them
+        self.request["BODY"] = parse.unquote(self.request.form.get("query", "{}"))
```

Comment:
This is so ugly. I understand, fast results and so, but this has to be fixed in a sane way at some point.